arbitrum-helpers/deployment.ts

```diff
@@ -0,0 +1,36 @@
+import { ContractFactory, Signer } from 'ethers'
+import { ethers } from 'hardhat'
+import { isEmpty } from 'lodash'
+
+import { waitForTx } from '.'
+
+export async function deployUsingFactory<T extends ContractFactory>(
+  signer: Signer,
+  factory: T,
+  args: Parameters<T['deploy']>,
+): Promise<ReturnType<T['deploy']>> {
+  const contractFactory = new ethers.ContractFactory(factory.interface, factory.bytecode, signer)
+  const contractDeployedCode = contractFactory.getDeployTransaction(...(args as any))
```

Comment:
This variable should maybe be named `contractInitCode` to avoid any confusion with deployed bytecode?